pkg/store/bucket.go

```diff
@@ -2132,6 +2159,17 @@ func (r *bucketIndexReader) fetchPostings(ctx context.Context, keys []labels.Lab
 		return uint64(ptrs[i].ptr.Start), uint64(ptrs[i].ptr.End)
 	})
 
+	if bytesLimiter != nil {
```

Comment:
What about having a `nopLimiter` to avoid nil checks?

Reply:
I thought it would be cleaner to add checks instead of passing `NewBytesLimiterFactory(0)()` everywhere :smile: let me do this change

Reply:
If we make the bytesLimiter an interface, then we don't need to pass the factory around right?

Reply:
But it is an interface already, no? I don't get your point. A `nopLimiter` is the same as `NewBytesLimiterFactory(0)()` so I'm not sure what is the benefit of having `nopLimiter`. Problem is that you have to pass some kind of limiter in _all_ places in all tests even though it is irrelevant. The purpose of this `if` is to avoid having to copy/paste that. Whether it is `nopLimiter` or `NewBytesLimiterFactory(0)()` it doesn't matter. Either way, let me make this change

Reply:
Thought of a better idea - since it will be pretty much necessary to pass something here, I've made it into a "normal" argument to `NewBucketStore`. With this, it won't be needed to add a bunch of `if`s everywhere.